network.c

```diff
@@ -54,8 +54,59 @@
 #ifndef MSG_NOSIGNAL
 #define MSG_NOSIGNAL 0
 #endif
+
+static int translate_wsa_error_to_posix(int wsa_err)
+{
+	switch (wsa_err) {
+	case WSAEACCES:          return -EACCES;
+	case WSAEADDRINUSE:      return -EADDRINUSE;
+	case WSAEADDRNOTAVAIL:   return -EADDRNOTAVAIL;
+	case WSAEAFNOSUPPORT:    return -EAFNOSUPPORT;
+	case WSAEALREADY:        return -EALREADY;
+	case WSAECONNREFUSED:    return -ECONNREFUSED;
+	case WSAECONNRESET:      return -ECONNRESET;
+	case WSAECONNABORTED:    return -ECONNABORTED;
+	case WSAEFAULT:          return -EFAULT;
+	case WSAEINPROGRESS:     return -EINPROGRESS;
+	case WSAEISCONN:         return -EISCONN;
+	case WSAEMFILE:          return -EMFILE;
+	case WSAENETDOWN:        return -ENETDOWN;
+	case WSAENETUNREACH:     return -ENETUNREACH;
+	case WSAENOBUFS:         return -ENOBUFS;
+	case WSAENOTSOCK:        return -ENOTSOCK;
+	case WSAEPROTONOSUPPORT: return -EPROTONOSUPPORT;
+	case WSAETIMEDOUT:       return -ETIMEDOUT;
+	case WSAEHOSTUNREACH:    return -EHOSTUNREACH;
+	case WSAEINVAL:          return -EINVAL;
+	case WSAEWOULDBLOCK:     return -EAGAIN;
+#ifdef ESHUTDOWN
+	case WSAESHUTDOWN:       return -ESHUTDOWN;
+#else
+	case WSAESHUTDOWN:       return -ECONNABORTED;
+#endif
+	case WSAENOTCONN:        return -ENOTCONN;
+	case WSAEOPNOTSUPP:      return -EOPNOTSUPP;
+	default: {
+		if (wsa_err > -4096 && wsa_err < 0) // pass through for POSIX errors
+			return wsa_err;
+		else
+			return -EIO;   // generic fallback
+	}
```

Comment:
DOo we need {} in here?

Reply:
not really. No local variable is being declared here so {} are actually not needed.